src/sr/discord_bot/bot.py

```diff
@@ -98,40 +98,64 @@ async def setup_hook(self) -> None:
 
     async def on_ready(self) -> None:
         self.logger.info(f"{self.user} has connected to Discord!")
-        guild = self.get_guild(self.guild.id)
-        if guild is None:
-            logging.error(f"Guild {self.guild.id} not found!")
+
+        if (guild := self.get_guild(self.guild.id)) is None:
+            self.logger.error("Guild not found: %r", self.guild.id)
+
+            # If the guild can't be refreshed, terminate here rather than continuing to
+            # validate known-stale data.
             exit(1)
+
         self.guild = guild
 
-        verified_role = discord.utils.get(guild.roles, name=VERIFIED_ROLE)
-        special_role = discord.utils.get(guild.roles, name=SPECIAL_ROLE)
-        volunteer_role = discord.utils.get(guild.roles, name=VOLUNTEER_ROLE)
-        supervisor_role = discord.utils.get(guild.roles, name=TEAM_LEADER_ROLE)
-        welcome_category = discord.utils.get(guild.categories, name=WELCOME_CATEGORY_NAME)
-        announce_channel = discord.utils.get(guild.text_channels, name=ANNOUNCE_CHANNEL_NAME)
-        feed_channel = discord.utils.get(guild.text_channels, name=FEED_CHANNEL_NAME)
-
-        if (
-            verified_role is None
-            or special_role is None
-            or volunteer_role is None
-            or supervisor_role is None
-            or welcome_category is None
-            or announce_channel is None
-            or feed_channel is None
-        ):
-            logging.error("Roles and channels are not set up")
-            exit(1)
+        setup_correctly = True
+
+        if (verified_role := discord.utils.get(guild.roles, name=VERIFIED_ROLE)) is None:
+            self.logger.error("Unable to find role: %r", VERIFIED_ROLE)
+            setup_correctly = False
         else:
             self.verified_role = verified_role
+
+        if (special_role := discord.utils.get(guild.roles, name=SPECIAL_ROLE)) is None:
+            self.logger.error("Unable to find role: %r", SPECIAL_ROLE)
+            setup_correctly = False
+        else:
```

Comment:
Hrm, this feels quite verbose and easy to slip an error into. I wonder if there's a better way to spell this. I'm _tempted_ to explore exception groups for this, though I don't have experience to know if that's actually viable here.

The assignment to specific instance members is the part where this is a touch more complicated, though I also wonder if a loop (building a mapping) would be useful here?

I don't know that this is a blocker, just wondering if there's a nicer way we can spell this while we're working on it.

Reply:
You'd still need to check each one (it's a null check, not an exception catch), log it, and append a value to _something_. You could save 1 line per if instead of `setup_correctly`, it was a list of strings which were when logged out, but I avoided that just in case there was an issue between calls, as it makes it easier to track.

I feel very strongly that it should not be a blocker.